lib/driver/pg.js

```diff
@@ -290,8 +290,21 @@ var PgDriver = Base.extend({
         } else {
           sql = util.format('ALTER TABLE "%s" ALTER COLUMN "%s" DROP DEFAULT', tableName, columnName);
         }
+        return this.runSql(sql).nodeify(setType.bind(this));
+      }
 
-        return this.runSql(sql).nodeify(callback);
+      function setType(err) {
+        if (err) {
+          return Promise.reject(err).nodeify(callback);
+        }
+        if (columnSpec.type != undefined){
+          var using = columnSpec.using != undefined ? 
+            columnSpec.using : util.format("USING %s::%s", columnName, this.mapDataType(columnSpec.type))
+          var sql = util.format('ALTER TABLE %s ALTER COLUMN "%s" TYPE %s %s', tableName, columnName, this.mapDataType(columnSpec.type), using);
+          return this.runSql(sql).nodeify(callback);
+        }
+        //should this have explicit else? 
+        return callback.call(this);
```

Comment:
Why do you call the callback directly? 
You should always return a promise insteada nd nodeify for the callback usage for backwards compatibility.